ArduPlane/AP_Arming.cpp

```diff
@@ -116,6 +116,12 @@ bool AP_Arming_Plane::pre_arm_checks(bool display_failure)
         ret = false;
     }
 
+    if (plane.quadplane.enabled() && ((plane.quadplane.options & QuadPlane::OPTION_ONLY_ARM_IN_QMODE_OR_AUTO) != 0) &&
+            !plane.control_mode->is_vtol_mode() && (plane.control_mode != &plane.mode_auto)) {
+        check_failed(display_failure,"not in Q mode");
```

Comment:
Really starting to look like we need a `pre_arm_check` on mode.  Copter has gotten away without that so far, but....